language/libra-vm/src/libra_vm.rs

```diff
@@ -273,6 +277,37 @@ impl LibraVM {
         Ok(())
     }
 
+    fn resolve_type_argument(
+        &self,
+        ctx: &mut SystemExecutionContext,
+        tag: &TypeTag,
+    ) -> VMResult<Type> {
+        let res = match tag {
+            TypeTag::U8 => Type::U8,
+            TypeTag::U64 => Type::U64,
+            TypeTag::U128 => Type::U128,
+            TypeTag::Bool => Type::Bool,
+            TypeTag::Address => Type::Address,
+            TypeTag::Vector(tag) => Type::Vector(Box::new(self.resolve_type_argument(ctx, tag)?)),
+            TypeTag::Struct(struct_tag) => {
+                let module_id = ModuleId::new(struct_tag.address, struct_tag.module.clone());
+                let ty_args = struct_tag
+                    .type_params
+                    .iter()
+                    .map(|tag| self.resolve_type_argument(ctx, tag))
+                    .collect::<VMResult<Vec<_>>>()?;
+                Type::Struct(Box::new(self.move_vm.resolve_struct_def_by_name(
+                    &module_id,
+                    &struct_tag.name,
+                    ctx,
+                    &ty_args,
+                )?))
+            }
+        };
+
+        Ok(res)
```

Comment:
Small nit, I like Putting the `Ok` around the whole match, then you don't need the temp. It might be weird, but I think it draws less attention to the `Ok` and I find it easier to read (but that might just be me)